### ISSUE TITLE
Split collision detection into snake collision and wall collision

### ISSUE THREAD

**Issue by clemens-tolboom** [ermiyaeskandary/Slither.io-bot#284]
Depends on #275

As found in #275 we should split the detection into two parts. That way we could use the scheduler for two separate task which can be toggles on/off.

This way we or our users can create snakes colliding into walls or change collision into snake behavior without missing the wall.

**Comment by ermiyaeskandary** [comment]
> create snakes colliding into walls

Magnificent with the movetoxy ! We could quickly get the bot to a long length and then test bots head to head or .... !
The skies the limit...
But it may be considered by cheating as some - I classify it as using resources at hand for educational purposes ;)

**Comment by ChadSki** [comment]
> Split collision detection into snake collision and wall collision

Why?

> That way we could use the scheduler for two separate task which can be toggles on/off.

Why would you ever want to collide into other snakes or the wall?

This sounds like over-engineering. Your scheduler is interesting but be wary of golden hammer syndrome (if all you have is a hammer everything looks like a nail).
